### ISSUE TITLE
[selectors] Multi-value attribute selectors

### ISSUE THREAD

**Issue by valtlai** [w3c/csswg-drafts#6042]
Spec: [Selectors](https://drafts.csswg.org/selectors-4/#attribute-selectors)

When styling text fields, there are many attribute values to be listed. I’m wondering could there be a syntax for listing multiple attribute values without repeating the attribute name. It could look something like this:

```css
input:not([type=button, checkbox, color, file, hidden, image, radio, range, reset, submit])
```

which would equal the following:
```css
input:not([type=button], [type=checkbox], [type=color], [type=file], [type=hidden],
          [type=image], [type=radio], [type=range], [type=reset], [type=submit])
```

For this specific use case, a pseudo-class like `:text-input` may be better, because listing input types is error prone and a particular input type may be be a text field or not depending on the browser and OS (for example `date` is a text field in major desktop browsers but a button in Chrome for Android). But there may be other use cases for attribute value lists.

(The original idea is not mine, but I can’t find the comment where this idea was mentioned.)

**Comment by SebastianZ** [comment]
The idea is obviously not new, though I like it.

As far as I can see, commas are allowed to separate [ident tokens](https://drafts.csswg.org/css-syntax-3/#ident-token-diagram). Though it needs to be clarified how string tokens are handled because they may contain commas.

So I guess the following two should be the same:
```css
span[data-x^=foo, bar]
```

```css
span[data-x^="foo", "bar"]
```

I.e. it matches all `<span>` elements with a `data-x` attribute value starting with either `foo` or `bar`.

Though the following is different:

```css
span[data-x^="foo, bar"]
```
Matching all `<span>` elements with a `data-x` attribute value starting with `foo, bar`.

And what I implied in these examples is that those value lists should be allowed for all attribute selectors.

Sebastian

**Comment by valtlai** [comment]
@SebastianZ Yeah, I totally agree with you. That’s how I think the syntax should work. Also, I don’t see a reason why not to allow the value lists for all types of attribute selectors.

**Comment by svgeesus** [comment]
Re-tagging requests for new functionality to Selectors 5, as Selectors 4 is being stabilized for CR Snapshot.
